ros_ws/src/autonomous/src/wall_following.cpp

```diff
@@ -136,7 +137,18 @@ void WallFollowing::lidarCallback(const sensor_msgs::LaserScan::ConstPtr& lidar)
 int main(int argc, char** argv)
 {
     ros::init(argc, argv, "wall_following");
-    WallFollowing wall_following;
+    ros::NodeHandle private_node_handle("~");
+
+    std::string frame_id;
+    private_node_handle.getParam(PARAMETER_MARKER_FRAME_ID, frame_id);
+    if (frame_id != MARKER_FRAME_ID_SIMULATION_VALUE && frame_id != MARKER_FRAME_ID_CAR_VALUE)
+    {
+        ROS_WARN_STREAM(PARAMETER_MARKER_FRAME_ID << " should be '" << MARKER_FRAME_ID_SIMULATION_VALUE << "' or '"
+                                                  << MARKER_FRAME_ID_CAR_VALUE << "'. Your value: '" << frame_id
+                                                  << "', using '" << MARKER_FRAME_ID_SIMULATION_VALUE << "'.");
+        frame_id = MARKER_FRAME_ID_SIMULATION_VALUE;
+    }
+    WallFollowing wall_following(frame_id);
```

Comment:
In our other nodes, the parameters are read in within the node class, not in the main function. I'll leave it to you to decide if it should be changed.
If the parameters are read inside the class, I think a default constructor for the `RvizGeometryPublisher` is needed.

Reply:
At first I also thought about putting it in the constructor, but for two reasons I did not do so.

1. There are (IMO) no always valid default values for the arguments given to the current constructor of the `RvizGeometryPublisher` class. So to whatever a default constructor would set the members, it would be in almost any case the wrong values. By not having a default constructor a `RvizGeometryPublisher` object can't be created by accident with invalid values, but only by actively give it the arguments to the constructor thus make it more robust, easier to debug and (IMO) less prone for errors.
2.  Having the constructor reading the parameter would make the class `WallFollowing` not be oblivious of the context in which is was constructed. Which would mean, that a (valid) `WallFollowing` object could only be created if it was startet in a ROS Launch context. It would also mean, that having more then one instance might leads to a hard to debug (and probably wrong) behavior.